content/4_Gold/Tree_Euler.mdx

```diff
@@ -715,6 +715,82 @@ preprocessing and query in $\mathcal{O}(1)$ time.
 
 <FocusProblem problem="rmqSample" />
 
+The following is an example implementation of a sparse table and code that
+answers LCA queries. Build time is $\mathcal{O}(N\log N)$, and queries are
+$\mathcal{O}(1)$.
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+template <typename T> struct RMQ {
+	int n, log2dist;
```

Comment:
make class, rename to `SparseTable`, and make vars private

Reply:
aight bet, gimme a sec